galasa-managers-parent/galasa-managers-internal-parent/dev.galasa.sem.manager/src/main/java/dev/galasa/sem/internal/SemManagerImpl.java

```diff
@@ -426,6 +440,20 @@ private void generateComplex() throws SemManagerException {
             if (rc > 4) {
                 throw new SemManagerException("SEM complex generation failed, rc=" +rc);
             }
+            
+            String loadLibrary = zosManager.getRunDatasetHLQ(primaryZosImage) + "." + getFramework().getTestRunName() + ".LOAD";
+            IZosDataset loadLibaryDs = zosFileManager.getZosFileHandler().newDataset(loadLibrary, primaryZosImage);
+            
+            // Add the run load library to DFHRPL            
+            if (loadLibaryDs.exists()) {
```

Comment:
A potential issue. If the ZosProgram manager is the manager that creates the DS for the load lib during the compile steps, then this is going to ignore the load library correct?

Reply:
No. If a load library is created, by this stage, it should exit.